datalad/local/export_archive.py

```diff
@@ -115,18 +110,20 @@ def _filter_tarinfo(ti):
                 compression) if archivetype == 'tar' else '')
 
         default_filename = "datalad_{.id}".format(dataset)
+        if filename is not None:
+            filename = Path(filename)
         if filename is None:
-            filename = default_filename  # in current directory
-        elif path.exists(filename) and path.isdir(filename):
-            filename = path.join(filename, default_filename) # under given directory
-        if not filename.endswith(file_extension):
-            filename += file_extension
+            filename = Path(default_filename)  # in current directory
```

Comment:
FTR: Was wondering whether `default_filename` should become Path already on line 112, but this rest of the logic seems to encourage not making this change.